[leong huen weng] iP

leonghuenweng opened the pull request.

leonghuenweng committed:
implemented greet function

leonghuenweng committed:
implemented greet, echo, exit functions

leonghuenweng committed:
Implemented Add, List function

leonghuenweng committed:
implemented "mark as done" function

Khulon commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,12 @@
+    public String name = new String();
+    public int number;
+
+    public Task(String namaewa,int num){
```
Naming should follow convention, use English

Khulon replied:
"num" should be written in full to be more clear (for better readability)

Khulon commented on src/main/java/ToDoList.java:
```diff
@@ -0,0 +1,51 @@
+import java.util.ArrayList;
+
+public class ToDoList {
+    private ArrayList<Task> TaskArrayList = new ArrayList<Task>();
```
"TaskArrayList" names should follow camelCase naming convention

Khulon commented on src/main/java/ToDoList.java:
```diff
@@ -0,0 +1,51 @@
+
+    public void printList(){
+        System.out.println(dash+"\nHere are the tasks in your list:\n");
+        for(int index = 0;index<TaskArrayList.size();index++){
```
index could be further expanded on to make it more clear which index it is referring to (better readability)

Khulon commented on src/main/java/ToDoList.java:
```diff
@@ -0,0 +1,51 @@
+
+    public void mark(Task item){
+        if(item.isDone){
+            item.isDone = false;       //unmark
```
Comments should be indented relative to their position in the code.

Khulon reviewed (commented):
Overall, code is clear and easy to understand! Just a few suggestions and pointers!

wuzhzn commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,37 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String dash = "__________________________________";
```
The string Dash can be renamed to all caps as it is a constant

wuzhzn commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,37 @@
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String dash = "__________________________________";
+        String greet = dash + "\nHello I'm Duke\nWhat can I do for you?\n" + dash;
```
Can possibly extract most of the methods to print user output to a separate class/method for better readability

wuzhzn commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,37 @@
+            String input = scan.nextLine();
+            String[] Temp = input.split(" ");
+            if(Temp[0].equals("mark")|Temp[0].equals("unmark")){
+                int numberToMark = Integer.parseInt(Temp[1]);
```
The naming can be better. In this case, since it's the index of the task, something like taskIndex might offer better readability.

wuzhzn commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,37 @@
+        while (true) {
+            Scanner scan = new Scanner(System.in);
+            String input = scan.nextLine();
+            String[] Temp = input.split(" ");
```
This is a temporary string array, so should not be named with caps

wuzhzn commented on src/main/java/ToDoList.java:
```diff
@@ -0,0 +1,51 @@
+        System.out.println(dash+"\nHere are the tasks in your list:\n");
+        for(int index = 0;index<TaskArrayList.size();index++){
+            if(TaskArray[index].isDone){
+                System.out.println(Integer.toString(index+1)+". [X]"+TaskArray[index].name+"\n");
```
The if-else conditions only differ by the [] and [X]. Might be more concise to summarise it in one statement through utilising the isDone method.

wuzhzn commented on src/main/java/ToDoList.java:
```diff
@@ -0,0 +1,51 @@
+        }
+        else{
+            item.isDone = true;       //mark
+            System.out.println(dash+"\nNice! I've marked this task as done:\n  [X] "+item.name);
```
Might be more readable to include/delete relevant spaces before and after the + or " "

wuzhzn reviewed (commented):
I like how the main method is quite abstract, and the code is generally quite good, with some mistakes in spaces and naming conventions.

leonghuenweng committed:
implemented deadline, event and to do classes

leonghuenweng committed:
changed Duke to Goot

leonghuenweng committed:
implemented exception handlers for deadlines and events

leonghuenweng committed:
Merge branch 'branch-Level-5'

leonghuenweng committed:
Implemented delete function, fixed validateInput function

leonghuenweng committed:
implemented save function for ToDo object, shifted GootExceptionHandler.validateInput to "else"  block in Goot.java file

leonghuenweng committed:
Implemented save for deadline and events

leonghuenweng committed:
task numberings are now updated whenever deletion occurs, also minor fixes implemented to handle invalid keywords

leonghuenweng committed:
Merge branch 'branch-Level-7'

leonghuenweng committed:
Merge branch 'branch-Level-6'

# Conflicts:
#	src/main/java/Goot.java
#	src/main/java/Task.java

leonghuenweng committed:
automated the save function (idk why I created my own save keyword )

leonghuenweng committed:
minor fixes to readBy function of deadline, it can now read everything behind "/by"

slightlyharp commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,43 @@
+
+    public Deadline(String taskName,int taskNumber,String by,String type){
+        super(taskName,taskNumber);
+        dueDate = by;
```
Consider use noun for variable. It is ok to use the same naming for the constructor parameter and the class variable.

slightlyharp commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,43 @@
+        //["deadline return book","by Sunday"]
+        String[] splitByForwardSlash = input.split("/");
+        //["by","Sunday"]
+//        String[] splitBySpace = splitByForwardSlash[1].split(" ");
```
Remove commented out code.

slightlyharp commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,43 @@
+        return (splitByForwardSlash[0].substring(9)).trim();
+    }
+    public static String readBy(String input){
+        //["deadline return book","by Sunday"]
```
Perhaps make a clearer explanation of the comment.
Do not write comments as if they are private notes to yourself. Instead, write them well enough to be understood by another programmer.

slightlyharp commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,53 @@
+import java.util.Arrays;
+
+public class Event extends Task{
+    private String from=new String();
```
Consider standardise the spacing before and after equal sign for better readability

slightlyharp commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,53 @@
+        //["event project meeting","from Mon 2pm ","to 4pm"]
+        String[] splitByForwardSlash = input.split("/");
+        //"Mon 2pm"
+        String fromString = splitByForwardSlash[1].substring(5).trim();
```
Consider avoid using magic number

slightlyharp commented on src/main/java/GootExceptions.java:
```diff
@@ -0,0 +1,3 @@
+public class GootExceptions extends Exception{
```
Consider using a more specific name for your exception

slightlyharp commented on src/main/java/Goot.java:
```diff
@@ -0,0 +1,83 @@
+    try{
+        Task.loadFileContentsToTaskArray(f);
+    }
+    catch (IOException e){
```
Perhaps it is better to catch the more specific exception. In this case use  FileNotFoundException instead of IOException since you only plan to catch FileNotFoundException.

slightlyharp commented on src/main/java/Goot.java:
```diff
@@ -0,0 +1,83 @@
+        Task.loadFileContentsToTaskArray(f);
+    }
+    catch (IOException e){
+        e.printStackTrace();
```
Consider print a user friendly text as well to explain the error to user

slightlyharp commented on src/main/java/Goot.java:
```diff
@@ -0,0 +1,83 @@
+        String input = scan.nextLine();
+        String[] inputSplitBySpace = input.split(" ");
+
+        if(inputSplitBySpace[0].equals("todo")){
```
Consider using switch instead of if else

slightlyharp commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,130 @@
+    //class level attributes
+    private static Task[] taskArray = new Task[100];
+    public static int lastIndex = 0;
+    public static String DASH = "__________________________________";
```
Consider use 'final' modifier for constant

slightlyharp commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,130 @@
+
+public class Task{
+    //class level attributes
+    private static Task[] taskArray = new Task[100];
```
Consider placing this variable in the Goot class instead

slightlyharp reviewed (commented):
Nice job! You might want to consider divide your files into packages. Remember to clean up your code, do not leave the commented out code there. You should also consider improve the level of abstractions, some methods are quite huge. And you can consider improve on the OOP structure for the up coming increment.

leonghuenweng committed:
1. hardcoded the file path for read and write file
2. settled the corner case of deleting list when there's only 1 item on the list

leonghuenweng committed:
refactored and categorised code into userInterface,Storage and Tasklist classes

leonghuenweng committed:
converted taskArray into private array

leonghuenweng committed:
implemented find function

leonghuenweng committed:
added JavaDoc and some minor bug fixes

leonghuenweng committed:
Merge branch 'branch-A-JavaDoc'

leonghuenweng committed:
Merge branch 'master' into branch-Level-9

leonghuenweng committed:
Merge pull request #1 from leonghuenweng/branch-Level-9

implemented find function

leonghuenweng committed:
Merge branch 'master' of https://github.com/leonghuenweng/ip

leonghuenweng committed:
created User Guide in README.md file and separated the userInterface.savedMessage() method from the Storage.save() method

leonghuenweng committed:
removed redundant code in README.md

leonghuenweng committed:
used relative path instead of absolute path for file creation, modified user interface

leonghuenweng committed:
added an Exception to prevent IndexOutOfBound error, and added bug fixes to delete function

leonghuenweng committed:
minor fixes in deleteMessage function

leonghuenweng committed:
included file creation details in user guide

leonghuenweng committed:
implemented exceptions for empty string/ whitespaces following keywords like delete, event etc
packaged classes
standardised the code
